File: src/platform/android/app/src/main/java/com/sgowen/gowenginedemo/Engine.kt
package com.sgowen.gowenginedemo

import android.content.res.AssetManager

class Engine
{
    init
    {
        init()
    }

    external fun createDeviceDependentResources(assetManager: AssetManager)
    external fun onWindowSizeChanged(screenWidth: Int, screenHeight: Int, cursorWidth: Int, cursorHeight: Int)
    external fun releaseDeviceDependentResources()
    external fun onResume()
    external fun onPause()
    external fun update(deltaTime: Double): Int
    external fun render()
    external fun onCursorDown(x: Float, y: Float)
    external fun onCursorDragged(x: Float, y: Float)
    external fun onCursorUp(x: Float, y: Float)

    private external fun init()

    companion object
    {
        init
        {
            System.loadLibrary("native-lib")
        }
    }
}

File: src/platform/apple/iOS/Engine.swift
//
//  Engine.swift
//  GowEngineDemo
//
//  Created by Stephen Gowen on 4/27/20.
//  Copyright © 2020 Stephen Gowen. All rights reserved.
//

import UIKit

public class Engine: NSObject
{
    var engine: NSEngine?
    
    public override init()
    {
        engine = NSEngine()
    }
    
    public func createDeviceDependentResources()
    {
        engine!.createDeviceDependentResources()
    }
    
    public func onWindowSizeChanged(screenWidth: Int32, screenHeight: Int32, cursorWidth: Int32, cursorHeight: Int32)
    {
        engine!.onWindowSizeChanged(screenWidth, screenHeight, cursorWidth, cursorHeight)
    }
    
    public func releaseDeviceDependentResources()
    {
        engine!.releaseDeviceDependentResources()
    }
    
    public func onResume()
    {
        engine!.onResume()
    }
    
    public func onPause()
    {
        engine!.onPause()
    }
    
    public func update(deltaTime: Double)
    {
        engine!.update(deltaTime)
    }
    
    public func render()
    {
        engine!.render()
    }
    
    public func onCursorDown(x: Float32, y: Float32, isAlt: Bool)
    {
        engine!.onCursorDown(x, y, isAlt)
    }
    
    public func onCursorDragged(x: Float32, y: Float32, isAlt: Bool)
    {
        engine!.onCursorDragged(x, y, isAlt)
    }
    
    public func onCursorUp(x: Float32, y: Float32, isAlt: Bool)
    {
        engine!.onCursorUp(x, y, isAlt)
    }
}
